### PR TITLE
Remove readOnly attrs from Clients Count Config Serializer

### PR DESCRIPTION
**Pull request by Monkeychip**
I noticed in the flashMessage displayed after saving the config that the three attrs were mentioned. These are readOnly and thus do not need to be serialized. This is related to the [census](https://github.com/hashicorp/vault/pull/20125) config work. 

While I was in there, I "octane'ified" the serializer.

image

Backporting to last three supported versions, which misses the 1.11 which had this change. But this was not a functional bug, just cosmetic, so I think that is okay.

### CODE REVIEW

File: ui/app/serializers/clients/config.js [hunk 1]
```diff
@@ -5,24 +5,27 @@
...
+export default class ClientsConfigSerializer extends ApplicationSerializer {
+  // these attrs are readOnly
+  attrs = {
+    billingStartTimestamp: { serialize: false },
+    minimumRetentionMonths: { serialize: false },
+    reportingEnabled: { serialize: false },
+  };
+
   normalizeResponse(store, primaryModelClass, payload, id, requestType) {
-    if (!payload.data) {
```
Comment (Monkeychip) [inline ui/app/serializers/clients/config.js:10]: @chelshaw I suspect this safe to remove, but thought I should check with you first? It was added 3ish years ago.

File: ui/app/serializers/clients/config.js [hunk 2]
```diff
@@ -5,11 +5,17 @@
...
+  // these attrs are readOnly
+  attrs = {
+    billingStartTimestamp: { serialize: false },
+    minimumRetentionMonths: { serialize: false },
+    reportingEnabled: { serialize: false },
+  };
+
   normalizeResponse(store, primaryModelClass, payload, id, requestType) {
     if (!payload.data) {
-      // CBS TODO: Remove this if block once API is published
```
Comment (Monkeychip) [inline ui/app/serializers/clients/config.js:11]: This is actually needed, we confirmed

File: ui/app/serializers/clients/config.js [hunk 3]
```diff
@@ -5,11 +5,17 @@
...
-      return this._super(store, primaryModelClass, payload, id, requestType);
+      return super.normalizeResponse(store, primaryModelClass, payload, id, requestType);
```
Comment (hellobontempo) [inline ui/app/serializers/clients/config.js:18]: could this also be the following? no need to change, just for my personal knowledge
```suggestion
      return super.normalizeResponse(...arguments);
```

**Review by hellobontempo — APPROVED** [review]
Thanks for tackling!

**Review by zofskeez — COMMENTED** [review]
Thanks for picking this up and modernizing the serializer! 🎉

**Pull request merged**